specs/DateTimePicker-Visual-Updates-Spec.md

```diff
@@ -0,0 +1,235 @@
+
+> Issue: are there other restrictions? Is there an underlying Comp page that this can point to?
+
+If the `ReplacementColor` property is set, the `SourceElement` is treated as a mask;
+every non-transparent pixel will be replaced by this color.
+
+
+The image below shows another use of the `ColorFilterOverlayControl` API, where it manipulates the colors of an element on screen where the contents of that element are moving. Here, the control is overlaid on a ScrollViewer that has a fixed position but moving content. 
+
+### ElementBackgroundGrid examples
+
+The following example puts an `ElementBackgroundGrid` on top of (z-ordered above)
+some scrolling text, causing the center of the text to be highlighted, even as it's scrolling.
+Things to note:
+* The `ElementBackgroundGrid` is sourcing from the `TextBlock`, it's also z-ordered above
+the same `TextBlock`. If the `Background` and `SourceElement` properties weren't set,
+it would have no effect; it would render the `TextBlock` exactly on top of itself.
+* The 'Background` property is black, and the `ReplacementColor` is green, causing
+the overlay text to be displayed as green on black.
+* The `ElementBackgroundGrid` is positioned and given a height such that it's a black band
+across the center of the scrolling text.
+
+
+```xml
+<Grid Background="White" Width="150">
+    <ScrollViewer x:Name="Target" Height="300">
+        <TextBlock
+            Foreground="Black"
+            FontSize="30"
+            Text="{x:Bind LoremIpsum}"
+            TextWrapping="Wrap">
+        </TextBlock>
+    </ScrollViewer>
+
+    <ElementBackgroundGrid
+        ReplacementColor="Green"
+        Background="Black"
+        TargetElement="{Binding ElementName=Target}"
+        Height="30"
+        VerticalAlignment="Center"
+        HorizontalAlignment="Stretch"/>
+</Grid>
+```
+
+![Example showing the ColorFilterOverlay control overlaid on moving content within a ScrollViewer, changing the foreground color and background color of words that pass through.](images/api-example.gif)
+
+The example below shows how the new `ColorFilterOverlayControl` API can be used to partially highlight a TextBlock.
+
+```xml
+ <Grid Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
+        <StackPanel x:Name="Panel" Orientation="Vertical" HorizontalAlignment="Center" VerticalAlignment="Center">
+            <Grid Margin="2,10,0,0">
+                <TextBlock x:Name="Target" Text="This string will be partially highlighted"/>
+                <ColorFilterOverlayControl
+                    ReplacementColor="Black"
+                    TargetElement="{Binding ElementName=Target}"
+                    Width="50"
+                    VerticalAlignment="Stretch"
+                    HorizontalAlignment="Center"
+                    Background="LightBlue"/>
+            </Grid>
+        </StackPanel>
+    </Grid>
+```
+
+> Todo: add a reflection example (classic use of VisualBrush in WPF)
```

Comment:
We don't have enough knobs to do this with the element as provided. In order to do that you would need to have the SourceRect / SourceOffset be available so that this can render offset from the original content. I mean ... ok I guess it would be possible with post-layout transforms, so maybe if you set RenderTransform on it you could do it? But I feel it would be somewhat unnatural.

Reply:
It's a standard use case of VisualBrush in WPF, it would be odd to suggest that you can't do the same thing in WinUI.

![Reflection](https://user-images.githubusercontent.com/18429489/111696178-3ae6bb80-87f1-11eb-882a-90e7f2b92e2d.png)